### PR TITLE
ttime: Remove TestTime

### PR DESCRIPTION
**Pull request by samuelkarp**
I'm attempting to remove the `TestTime` struct and the package-level time functions from `ttime` and replace with either integration tests that directly use the real `time` implementation or unit tests that use a mock.  This is still WIP for now.

r? @euank @aaithal @juanrhenals

### CODE REVIEW

File: agent/acs/handler/acs_handler.go [hunk 1]
```diff
@@ -69,6 +70,39 @@ type StartSessionArguments struct {
 	ECSClient            api.ECSClient
 	StateManager         statemanager.StateManager
 	AcceptInvalidCert    bool
+	_time                ttime.Time
```
Comment (aaithal) [inline agent/acs/handler/acs_handler.go]: It seems like you can put `_time` and `_timeOnce` and initialization logic into a struct. We can avoid a bunch of duplicated code with that I think.

File: scripts/generate/mockgen.sh [hunk 2]
```diff
@@ -24,7 +24,7 @@ export PATH="${GOPATH//://bin:}/bin:$PATH"
 
 data=$(
 cat << EOF
-// Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// Copyright 2015-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (aaithal) [inline scripts/generate/mockgen.sh]: While you're here, do you also want to just do this?

```
year="$(date +"%Y")"
Copyright 2015-${year}
```

**Comment by aaithal** [comment]
lgtm, i have a couple of minor comments.

File: agent/acs/handler/acs_handler.go (hunk 1 above)
Comment (samuelkarp) [inline agent/acs/handler/acs_handler.go]: I'd like to punt on this for now.  I think there's value in reusing the logic, but the better value is switching to proper dependency injection and removing the logic altogether.

Comment (aaithal) [inline agent/acs/handler/acs_handler.go]: alright, sgtm

File: scripts/generate/mockgen.sh (hunk 2 above)
Comment (samuelkarp) [inline scripts/generate/mockgen.sh]: Done.

**Comment by aaithal** [comment]
:shipit:

**Comment by euank** [comment]
Crazy time warp logic, you shall be missed :cry:

**Comment by sentientmonkey** [comment]
tumblr_n0hoj8hghz1rkxuejo1_500

**Pull request merged**